zospy/analyses/new/physicaloptics/physical_optics_propagation.py

```diff
@@ -0,0 +1,548 @@
+        else:
+            self.analysis.Settings.UsePeakIrradiance = True
+            self.analysis.Settings.PeakIrradiance = self.settings.peak_irradiance
+
+        if self.settings.beam_file != "":
+            self.analysis.Settings.BeamFile = self.settings.beam_file
+
+        self._set_pop_parameters("beam", self.settings.beam_parameters)
+
+        # Display settings
+        self.analysis.Settings.ShowAs = constants.process_constant(
+            constants.Analysis.PhysicalOptics.POPShowAsTypes, self.settings.show_as
+        )
+        self.analysis.Settings.DataType = constants.process_constant(
+            constants.Analysis.PhysicalOptics.POPDataTypes, self.settings.data_type
+        )
+        self.analysis.Settings.Project = constants.process_constant(
+            constants.Analysis.PhysicalOptics.POPProjectionTypes, self.settings.project
+        )
+
+        if self.settings.show_as == "Contour":
+            self.analysis.Settings.ContourFormat = self.settings.contour_format
+        elif self.settings.show_as in ("CrossX", "CrossY"):
+            self.analysis.Settings.RowOrColumn = (
+                0 if self.settings.row_or_column == "Center" else self.settings.row_or_column
+            )
+
+        self.analysis.Settings.PlotScale = self.settings.plot_scale
+        self.analysis.Settings.ScaleType = constants.process_constant(
+            constants.Analysis.PhysicalOptics.POPScaleTypes, self.settings.scale_type
+        )
+        self.analysis.Settings.ZoomIn = constants.process_constant(
+            constants.Analysis.PhysicalOptics.POPZoomTypes, self.settings.zoom_in
+        )
+        self.analysis.Settings.ZeroPhaseLevel = self.settings.zero_phase_level
+
+        self.analysis.Settings.SaveOutputBeam = self.settings.save_output_beam
+        if self.settings.save_output_beam:
+            self.analysis.Settings.OutputBeamFile = self.settings.output_beam_file
+            self.analysis.Settings.SaveBeamAtAllSurfaces = self.settings.save_beam_at_all_surfaces
+
+        # Fiber settings
+        self.analysis.Settings.ComputeFiberCouplingIntegral = self.settings.compute_fiber_coupling_integral
+
+        if self.settings.compute_fiber_coupling_integral:
+            self.analysis.Settings.IgnoreFiberPolarization = self.settings.ignore_fibre_polarization
+            self.analysis.Settings.FiberType = constants.process_constant(
+                constants.Analysis.PhysicalOptics.POPFiberTypes, self.settings.fiber_type
+            )
+            self.analysis.Settings.FiberPosition = constants.process_constant(
+                constants.Analysis.PhysicalOptics.POPFiberPositions, self.settings.fiber_position
+            )
+            self.analysis.Settings.TiltAboutX = self.settings.tilt_about_x
+            self.analysis.Settings.TiltAboutY = self.settings.tilt_about_y
+
+            if self.settings.fiber_type_file != "":
+                self.analysis.Settings.FiberTypeFile = self.settings.fiber_type_file
```

Comment:
@LucVV can you please check if your OpticStudio version indeed uses `FiberTypeFile` for this setting? OpticStudio 24R2 has `FiberTypeFilename`